### ISSUE TITLE
Common Form Design Causing Failure of 1.3.1 Info and Relationships

### ISSUE THREAD

**Issue by kimviens** [w3c/wcag#4041]
Hello everyone,

This conversation mainly starts from reading the following quote taken from [Form Instructions | Web Accessibility Initiative (WAI) | W3C](https://www.w3.org/WAI/tutorials/forms/instructions/) : 

> Important: Screen readers often switch to “Forms Mode” when they are processing content within a < form > element. In this mode they usually only read aloud form elements such as < input >, < select >, < textarea >, < legend >, and < label >. It is critical to include form instructions in ways that can be read aloud. This will be explained further below.

This seems to imply that if a form instruction is present without it being read aloud when a screen reader user is in "Forms Mode", then there might be a 1.3.1 Info and Relationships failure or another WCAG failure. I definitely stand by this for forms instructions like a message about the format of a field, or labels or other input hints. However, there are some instances where programmatically linking extra information would make the form fields overly verbose.

Let's talk about the use of headings and form sections. Here is an example:

```
<h2>Animal Expertise Competencies</h2>

<h3>Critical Thinking</h3>
<p>Evaluating and analyzing information to gain profound insights into diverse aspects of animal behavior, physiology, ecology, and evolution. This includes studying their social structures, communication patterns, reproductive strategies, and foraging behaviors. Critical thinking also involves assessing the impact of environmental factors on animal populations, such as climate change, habitat loss, and pollution. Furthermore, it involves interpreting scientific research, conducting field observations, and using statistical methods to draw meaningful conclusions about animal species and their interactions with the environment and other species.</p>

<label for="problem-resolution">Based on your current expertise and experience, how do you approach resolving complex challenges or incidents related to animals?</label>
<textarea id="problem-resolution"></textarea>

<h3>Client-Oriented Approach</h3>
<p>Identifying and addressing the unique needs of clients, while providing exceptional service in terms of animal knowledge. This includes assisting clients in understanding the behavior, care, and welfare of specific animal species, as well as providing guidance on appropriate habitat design, nutrition, and enrichment strategies. Additionally, a client-oriented approach involves offering advice on disease prevention, veterinary care, and responsible pet ownership. It also entails educating clients about wildlife conservation, sustainable practices, and ethical considerations related to animal interactions and activities.</p>

<label for="client-response">When a client approaches you with a request for animal-related information, how do you ensure a prompt and satisfactory response?</label>
<textarea id="client-response"></textarea>
```

Should this kind of example still be a candidate for a fieldset? Technically, the fields are grouped because they are grouped within headings. Also, with the lengthy p tag, its not an enjoyable experience to constantly hear that text when tabbing into the form fields.

And what if I tell you that my p tags are supplemental information such that it does not influence the data entry however, I insist on keeping my p tags because I want the supplemental information in case someone is interested in it.

**To conclude, I want to know if I need to file WCAG failures for any content that does not get read by the screen reader in forms mode when in a form.** I gave you an example of where you could find information that is not programmatically linked to the inputs.

**Comment by kimviens** [comment]
To restate the question: **Should any content that does not get read by a screen reader while you are tabbing in a form in "Forms mode" get flagged as a WCAG failure?**

**Comment by TestPartners** [comment]
In my view, as a general rule you should not base pass / fail results on the behaviour of assistive technologies. The results should be based on inspection of the user interface and source code. This is because you are testing for conformance with the success criteria, and you are not testing the user experience. There are a few exceptions, such as ARIA live regions, which can be very difficult to test without a screen reader, depending on how they are coded. SC 1.3.1 isn't one of these exceptions.

In your example, there is no need for a fieldset. There might be if those fields were part of a larger form, but even then it's unlikely. Fieldsets are for groups of related fields, such as radio button groups or address fields. I don't think the presence of headings is relevant.

You can programmatically link supplemental information to a form field using the "aria-describedby" attribute. However, that is usually done with small amounts of text such as data validation rules. It would not be appropriate in your case.

In practical terms, screen reader users can use a variety of in-page navigation approaches depending on their proficiency and priorities. If they want to be sure they don't miss anything, they can use virtual cursor mode to read all the content, only going into forms mode when they reach a form control. If speed is a priority, they may choose to tab through the form controls, but they know they may miss some important information. The author is not obliged to ensure that all information is presented to the user when they tab through the form.

**Comment by Hvychvy383** [comment]
![k830 2](https://github.com/user-attachments/assets/a5251741-45a0-46cc-8cde-e7ec9784940f)

**Comment by kimviens** [comment]
Hi @TestPartners, I agree when I am testing I am not basing myself off of adaptive technologies. However, often when an adaptive technology is not behaving correctly its often a code smell for a WCAG failure. 

I personally do not agree that supplemental information in a form that is not programmatically linked to an input is a failure. I am actually debating this with someone else as I agree with you:
> In practical terms, screen reader users can use a variety of in-page navigation approaches depending on their proficiency and priorities

I think I'll register this case as a best practice rather than a failure in my books. For sure it would be nice if a screen reader user could just tab through the form and get all of the information. But that is not always feasible.

Anyone else got something to add?

**Comment by kimviens** [comment]
Hi @TestPartners, I apologize for any confusion. I agree with everything you've said and yes, I am aware that you can create programmatic links if desired. 

This conversation began with a disagreement regarding someone's suggestion that if content in a form is not programmatically linked to an input, it is considered a 1.3.1 Info and Relationships issue. All that because of [Form Instructions | Web Accessibility Initiative (WAI) | W3C](https://www.w3.org/WAI/tutorials/forms/instructions/) :

> Important: Screen readers often switch to “Forms Mode” when they are processing content within a < form > element. In this mode they usually only read aloud form elements such as < input >, < select >, < textarea >, < legend >, and < label >. **It is critical to include form instructions in ways that can be read aloud. This will be explained further below.**

Their proposed solution to address this issue is to either create programmatic links between the information and the inputs or to place all supplementary information at the top of the form, before users start navigating through the form using the tab key.

I wish I could get more opinion from the community on this.

**Comment by kimviens** [comment]
Hey @patrickhlauke, I am not sure if I expressed the issue well enough in the comments above. But would you have a second to give your opinion on this?

The person I am disagreeing with fears that screen reader users will miss information in a form if its not programmatically linked. And it does not always make sense to use fieldsets and programmatic links.

**Comment by patrickhlauke** [comment]
There's often no hard yes/no cut-off point. But in general, I would say no, your scenario doesn't sound like a hard fail of 1.3.1. As a best practice, perhaps consider doing some extra wrapping with `<fieldset>` and `<legend>` (and having headings inside the latter to cover all bases), but I certainly wouldn't automatically fail a site for not doing it.

> fears that screen reader users will miss information

Not saying that this is the case here exactly, but I often find the attitude of "screen reader users are delicate and helpless little flowers that won't be able to deal with things if we don't spoon-feed them". While sometimes yes, it makes sense to go above and beyond basic compliance if it means a better user experience, it's also worth sometimes just reaching out to actual SR users for their opinion directly - keeping in mind that it's often only that, their specific personal opinion/preference, and that a single SR user does not necessarily represent *all* SR users either.

**Comment by JAWS-test** [comment]
Regardless of SC 1.3.1: For short notes it makes sense to link them to the form field via aria-describedby, for longer notes (as in your example) this does not make sense. Even the labels in your example are quite long.

**Comment by kimviens** [comment]
Hi @TestPartners, @patrickhlauke, @detlevhfischer and @JAWS-test , thank you so much for your opinions on this. I now feel good about this and therefore will close this issue.

#### In Summary: 

Not linking extra content to form fields does not always mean that it fails the requirement 1.3.1 Info and Relationships. If the failure to link the content prevents users from properly filling out the form, then it may be considered a failure (error messages). However, if the extra content includes headings and "mood setting" information, it does not necessarily constitute a failure.

Simply failing content because it is not heard by screen reader users in forms mode is not a satisfactory reason to declare it as a failure.
